auditbeat/module/file_integrity/kprobes/events_process.go

```diff
@@ -0,0 +1,240 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build linux
+
+package kprobes
+
+import (
+	"context"
+	"errors"
+	"path/filepath"
+
+	"golang.org/x/sys/unix"
+)
+
+type Emitter interface {
+	Emit(ePath string, pid uint32, op uint32) error
+}
+
+type eProcessor struct {
+	p           pathTraverser
+	e           Emitter
+	d           *dEntryCache
+	isRecursive bool
+}
+
+func newEventProcessor(p pathTraverser, e Emitter, isRecursive bool) *eProcessor {
+	return &eProcessor{
+		p:           p,
+		e:           e,
+		d:           newDirEntryCache(),
+		isRecursive: isRecursive,
+	}
+}
+
+func (e *eProcessor) process(ctx context.Context, pe *ProbeEvent) error {
```

Comment:
ctx is unused. Is that expected? If it is then perhaps remove the param.

Reply:
yes this is unused for now; it is essentially the monitor context and the initial idea was to have it there in case we need to escape a blocking writing/reading of a channel in case the monitor was instructed to stop; the initial implementation design has the event processor reading from a channel of path traverser to get the stated paths but this got changed throughout the implementation. Ofc I will mark it as unused but I still think that having a context in the event processor isn't that terrible of an idea, so if you are okay with it(?) I will leave it there for now

Reply:
I'm ok with leaving it. I would change it over to the blank identifier to indicate it was intentional.

Reply:
already changed to blank identifier 🙂